modules/cloudevent-pull-trigger/README.md

```diff
@@ -0,0 +1,122 @@
+# `cloudevent-pull-trigger`
+
+```mermaid
+flowchart LR
+    subgraph "regional network"
+    A[[Pub/Sub topic]] -- creates and inserts --> B(Subscription)
+    B(Subscription) -- pulls --> C(Consumer)
+    end
+```
+
+This module abstracts regionalized event-triggered services, but assumes the
+pull based consumption of events. This is very similar to the
+[`cloudevent-trigger`](./../cloudevent-trigger/), but better suited for very
+bursty traffic, where you can control the consumption of events by how you
+choose to pull them from the subscription. It's intended to be used with the
+"Broker" abstraction is described by the sibling
+[`cloudevent-broker`](./../cloudevent-broker/) module.
+
+The intended usage of this module for consuming events is something like this:
+
+```hcl
+// Create a network with several regional subnets
+module "networking" {
+  source = "chainguard-dev/common/infra//modules/networking"
+
+  name       = "my-networking"
+  project_id = var.project_id
+  regions    = [...]
+}
+
+// Create the Broker abstraction.
+module "cloudevent-broker" {
+  source = "chainguard-dev/common/infra//modules/cloudevent-broker"
+
+  name       = "my-broker"
+  project_id = var.project_id
+  regions    = module.networking.regional-networks
+}
+
+// Set up regionalized triggers to deliver filtered events from our regionalized
+// brokers to subscriptions.
+module "cloudevent-pull-trigger" {
+  for_each = module.networking.regional-networks
+
+  source = "chainguard-dev/common/infra//modules/cloudevent-pull-trigger"
+
+  name       = "bar"
+  project_id = var.project_id
+  broker     = module.cloudevent-broker.broker[each.key]
+
+  // Only trigger on bar-type events.
+  filter     = { "type" : "dev.chainguard.bar" }
+
+  // Only trigger on events with subject matching "foo/*"
+  filter_prefix = { "subject": "foo/" }
+}
+
+// Grant your subscriber the permissions to subscribe to trigger created above.
+resource "google_pubsub_subscription_iam_member" "pull_event_subscriber" {
+  subscription = module.cloudevent-pull-trigger.pull-subscription
+  role         = "roles/pubsub.subscriber"
+  member       = "serviceAccount:${var.cve_detection_sa_email}"
```

Comment:
I wonder if we want an abstraction like `authorize-pull-subscription` that handles this similar to: https://github.com/chainguard-dev/terraform-infra-common/tree/main/modules/authorize-private-service

I did this (instead of just plumbing the URL around) to encourage folks to use this to look up the Cloud Run URLs in a way that ensured things were authorized.  There is a dual here for the service accessing the pull subscription.

Reply:
It's only for the service account, seems cleaner to just do this since you create this likely from the same place as you create the subscription. We can talk about this more, but using it like this seems easy/simple enough.